### PR TITLE
Added release workflow

### PR DESCRIPTION
**Pull request by Imad-Allal**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: Create Release and Tag
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Version for the release"
+        required: true
+        default: "0.1.0"
```
Comment (lakonema2000) [inline .github/workflows/release.yml:9]: please remove this: we should read the version from a config file that is added to the repo.
chatgpt can advise about how to do that with go

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Create GitHub Release
+        uses: actions/create-release@v1
+        with:
+          tag_name: v${{ github.event.inputs.version }}
+          release_name: v${{ github.event.inputs.version }}
+          draft: false
+          prerelease: false
+          body: "Release of nova-sdk version v${{ github.event.inputs.version }}"
+
+      - name: Go to module directory
```
Comment (lakonema2000) [inline .github/workflows/release.yml:39]: and then? we should also push the project to pkg.go.dev.
look at [this page](https://pkg.go.dev/about#adding-a-package) to understand how to add our package there

Comment (Imad-Allal) [reply .github/workflows/release.yml]: No need for that. The Go module is automatically indexed by `pkg.go.dev` once the tagged version is pushed to the repository, and we are already creating a tag in the `.yml` file. You can check this [package](https://pkg.go.dev/github.com/Imad-Allal/pkg-go-dev-demo@v0.1.0) for instance.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+          draft: false
+          prerelease: false
+          body: "Release of nova-sdk version v${{ steps.read_version.outputs.VERSION }}"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Go to module directory and tidy up Go modules
+        run: |
+          cd go
+          go mod tidy
```
Comment (lakonema2000) [inline .github/workflows/release.yml:46]: let's remove that

Comment (lakonema2000) [inline .github/workflows/release.yml:46]: let's remove that part

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
+name: Create Release and Tag
```
Comment (lakonema2000) [inline .github/workflows/release.yml:1]: update the name

**Review by lakonema2000 — APPROVED** [review]

**Pull request merged**